modules/nextflow/src/main/groovy/nextflow/cli/CmdLineage.groovy

```diff
@@ -0,0 +1,291 @@
+        void apply(List<String> args)
+        void usage()
+    }
+
+    private List<SubCmd> commands = new ArrayList<>()
+
+    private LinCommand operation
+
+    private ConfigMap config
+
+    CmdLineage() {
+        commands << new CmdLog()
+        commands << new CmdDescribe()
+        commands << new CmdRender()
+        commands << new CmdDiff()
+        commands << new CmdFind()
+    }
+
+    @Parameter(hidden = true)
+    List<String> args
+
+    @Override
+    String getName() {
+        return NAME
+    }
+
+    @Override
+    void run() {
+        if( !args ) {
+            usage(List.of())
+            return
+        }
+        // setup the plugins system and load the secrets provider
+        Plugins.init()
+        // load the config
+        this.config = new ConfigBuilder()
+            .setOptions(launcher.options)
+            .setBaseDir(Paths.get('.'))
+            .build()
+        // init plugins
+        Plugins.load(config)
+        // load the command operations
+        this.operation = Plugins.getExtension(LinCommand)
+        if( !operation )
+            throw new IllegalStateException("Unable to load lineage extensions.")
+        // consume the first argument
+        getCmd(args).apply(args.drop(1))
+    }
+
+    /**
+     * Print the command usage help
+     */
+    void usage() {
+        usage(args)
+    }
+
+    /**
+     * Print the command usage help
+     *
+     * @param args The arguments as entered by the user
+     */
+    void usage(List<String> args) {
+        if( !args ) {
+            List<String> result = []
+            result << this.getClass().getAnnotation(Parameters).commandDescription()
+            result << "Usage: nextflow $NAME <sub-command> [options]".toString()
+            result << ''
+            result << 'Commands:'
+            int len = 0
+            commands.forEach {len = it.name.size() > len ? it.name.size() : len }
+            commands.sort(){it.name}.each { result << "  ${it.name.padRight(len)}\t${it.description}".toString()  }
+            result << ''
+            println result.join('\n').toString()
+        }
+        else {
+            def sub = commands.find { it.name == args[0] }
+            if( sub )
+                sub.usage()
+            else {
+                throw new AbortOperationException("Unknown $NAME sub-command: ${args[0]}")
+            }
+        }
+    }
+
+    protected SubCmd getCmd(List<String> args) {
+
+        def cmd = commands.find { it.name == args[0] }
+        if( cmd ) {
+            return cmd
+        }
+
+        def matches = commands.collect{ it.name }.closest(args[0])
+        def msg = "Unknown cloud sub-command: ${args[0]}"
+        if( matches )
+            msg += " -- Did you mean one of these?\n" + matches.collect { "  $it"}.join('\n')
+        throw new AbortOperationException(msg)
+    }
+
+    class CmdLog implements SubCmd {
+
+        @Override
+        String getName() {
+            return 'log'
```

Comment:
If we rename `describe` to `view`, maybe we should also rename `log` to `list` to align more with the pipeline commands, and to not confuse with `nextflow log` or `.nextflow.log`

Reply:
Fair enough